app/cdap/components/GridTable/index.tsx

```diff
@@ -0,0 +1,286 @@
+    };
+    getWorkSpaceData(payload, wid);
+  }, [wid]);
+
+  const createHeadersData = (columnNamesList: any, columnLabelsList, columnTypesList) => {
+    if (Array.isArray(columnNamesList)) {
+      return columnNamesList.map((eachColumnName) => {
+        return {
+          name: eachColumnName,
+          label: eachColumnName,
+          type: [columnTypesList[eachColumnName]],
+        };
+      });
+    }
+  };
+
+  const convertNonNullPercent = (key, nonNullValue) => {
+    const lengthOfData = gridData.values.length;
+    let count = 0;
+    let nonNull: any = 0;
+    let empty: any = 0;
+    let nullValue: any = 0;
+    if (lengthOfData) {
+      nonNull = nonNullValue['non-null'] ? (nonNullValue['non-null'] / 100) * lengthOfData : 0;
+      nullValue = nonNullValue.null ? (nonNullValue.null / 100) * lengthOfData : 0;
+      empty = nonNullValue.empty ? (nonNullValue.empty / 100) * lengthOfData : 0;
+      count = parseInt(nullValue + empty);
+    }
+    return count;
+  };
+
+  const checkFrequentlyOccuredValues = (key) => {
+    const valueOfKey = gridData.values.map((el) => el[key]);
+    let mostfrequentItem = 1;
+    let count = 0;
+    let item = '';
+    const data = {
+      name: '',
+      count: 0,
+    };
+    for (let i = 0; i < valueOfKey.length; i++) {
+      for (let j = i; j < valueOfKey.length; j++) {
+        if (valueOfKey[i] == valueOfKey[j]) {
+          count++;
+        }
+        if (mostfrequentItem < count) {
+          mostfrequentItem = count;
+          item = valueOfKey[i];
+        }
+      }
+      count = 0;
+      item = item == '' ? valueOfKey[i] : item;
+    }
+    data.name = item;
+    data.count = mostfrequentItem;
+    return data;
+  };
+
+  const createMissingData = (statistics) => {
+    const objectArray = Object.entries(statistics);
+    const metricArray = [];
+    objectArray.forEach(([key, value]) => {
+      const valueToArray = Object.entries(value);
+      const tempArray = [];
+      valueToArray.forEach(([vKey, vValue]) => {
+        tempArray.push({
+          label:
+            vKey == 'general' && convertNonNullPercent(key, vValue) == 0
+              ? checkFrequentlyOccuredValues(key).name
+              : vKey == 'general'
+              ? 'Missing/Null'
+              : vKey == 'types'
+              ? ''
+              : '',
+          count:
+            vKey == 'types'
+              ? ''
+              : convertNonNullPercent(key, vValue) == 0
+              ? checkFrequentlyOccuredValues(key).count
+              : convertNonNullPercent(key, vValue),
+        });
+      }),
+        metricArray.push({
+          name: key,
+          values: tempArray.concat(invalidCountArray),
+        });
+    });
+    return metricArray;
+  };
+
+  const getGridTableData = async () => {
+    const rawData: any = gridData;
+    const headersData = createHeadersData(rawData.headers, rawData.headers, rawData.types);
+    setHeadersNamesList(headersData);
```

Comment:
could this set it to undefined or something?

Reply:
done

Reply:
we have added a specific type for acceptable values, please let us know if this is fine. Otherwise plz elaborate the comment.